httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java

```diff
@@ -233,6 +231,8 @@ public Reader asReader(Charset charset) throws IOException {
       @Override
       public void close() throws IOException {
         EntityUtils.consume(entity);
+        if (httpResponse instanceof CloseableHttpResponse)
+          ((CloseableHttpResponse) httpResponse).close();
```

Comment:
could this close method call be enclosed in a `finally` block? that would ensure the socket is released even in the event of an exception when consuming the entity.
It is also the approach recommended here - https://hc.apache.org/httpcomponents-client-ga/quickstart.html
quoted text from the document 

> // In order to ensure correct deallocation of system resources
> // the user MUST call CloseableHttpResponse#close() from a finally clause.
>try{
>    EntityUtils.consume(entity1);
>} finally {
>    response1.close();
>}

-  Additionally it would also not show up as a potential issue in static code analyzers like Fortify, which expect to see a finally block which closes the connection.

Reply:
hi @velo you reckon, this finally block something we could incorporate in here?

Reply:
good point, lemme me do it

Reply:
That would be really great. Thank you @velo